exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillResultSetImpl.java

```diff
@@ -125,7 +154,7 @@ protected void cancel() {
   // (Not delegated.)
   @Override
   public boolean next() throws SQLException {
-    throwIfClosed();
+    throwIfTimedOutOrClosed();
```

Comment:
The query cancellation should take care of it. It'll be hard to have a unit test specifically for this, but I'll try.